debezium-core/src/main/java/io/debezium/pipeline/signal/channels/FileSignalChannel.java

```diff
@@ -0,0 +1,152 @@
+ * <li>{@code type STRING} - the unique logical name of the code executing the signal</li>
+ * <li>{@code data STRING} - the data in JSON format that are passed to the signal code
+ * </ul>
+ *
+ * @author Ismail Simsek
+ *
+ */
+public class FileSignalChannel implements SignalChannelReader {
+    public static final Queue<SignalRecord> SIGNALS = new ConcurrentLinkedQueue<>();
+    public static final String CONFIGURATION_FIELD_PREFIX_STRING = "signal.";
+    public static final Field SIGNAL_FILE = Field.create(CONFIGURATION_FIELD_PREFIX_STRING + "file")
+            .withDisplayName("Signal file name")
+            .withType(ConfigDef.Type.STRING)
+            .withWidth(ConfigDef.Width.LONG)
+            .withImportance(ConfigDef.Importance.HIGH)
+            .withDescription("The name of the file for the signals to the connector")
+            .withValidation(Field::isRequired);
+    public static final String CHANNEL_NAME = "file";
+    private static final Logger LOGGER = LoggerFactory.getLogger(FileSignalChannel.class);
+    ObjectMapper mapper = new ObjectMapper();
+    private File signalFile;
+
+    @Override
+    public String name() {
+        return CHANNEL_NAME;
+    }
+
+    @Override
+    public void init(CommonConnectorConfig connectorConfig) {
+
+        Configuration signalConfig = connectorConfig.getConfig().subset(CONFIGURATION_FIELD_PREFIX_STRING, false)
+                .edit()
+                .withDefault(FileSignalChannel.SIGNAL_FILE, "file-signals.txt")
+                .build();
+        this.signalFile = new File(signalConfig.getString(SIGNAL_FILE));
+        LOGGER.info("Reading '{}' file for signals", signalFile.getAbsolutePath());
+    }
+
+    @Override
+    public List<SignalRecord> read() {
+        List<SignalRecord> signals = new ArrayList<>();
+
+        if (!signalFile.exists() || signalFile.isDirectory()) {
+            LOGGER.trace("Signal file not found '{}'", signalFile.getAbsolutePath());
+            return signals;
+        }
+
+        try {
+            // read all the signals from file. signal line must be json formatted
+            List<String> lines = Files.readAllLines(signalFile.toPath());
+            if (!lines.isEmpty()) {
+                // remove signals from file by overriding file with empty content
+                new FileWriter(signalFile, false).close();
+            }
+            Iterator<String> lineIterator = lines.iterator();
+            while (lineIterator.hasNext()) {
+                String signalLine = lineIterator.next();
+                if (signalLine == null || signalLine.isBlank()) {
+                    LOGGER.debug("Ignoring empty signal line: `{}`", signalLine);
+                    lineIterator.remove();
+                    continue;
+                }
+                try {
+                    LOGGER.trace("Processing signal line: {}", signalLine);
+                    JsonNode signalJson = mapper.readTree(signalLine);
```

Comment:
What do you think extracting line 103-110 in a method to improve readability?

Reply:
updated